internal/pkg/docker/dockerengine/dockerengine.go

```diff
@@ -343,6 +346,13 @@ func (c DockerCmdClient) Run(ctx context.Context, options *RunOptions) error {
 			exec.Stdout(stdout),
 			exec.Stderr(stderr),
 			exec.NewProcessGroup()); err != nil {
+			var ec exitCodeError
+			if errors.As(err, &ec) {
```

Comment:
Where inside of `RunWithContext` will it return an `err` that is an `exitCodeError`?

Reply:
ah, i see its the `ExitError` from `os/exec`, within which there is an [ProcessState](https://pkg.go.dev/os#ProcessState)

Reply:
yeah I want to capture if the `docker run` cmd exits with exit code `0` which indicate container state as `success`.